petstore/src/test/scala/io/finch/petstore/PetstoreDbSpec.scala

```diff
@@ -0,0 +1,299 @@
+      Some(false))
+    val idFuture: Future[Long] = db.addOrder(mouseCircusOrder)
+    Await.result(idFuture.liftToTry).isThrow
+  }
+
+  //DELETE: Delete purchase order by ID
+  it should "be able to delete orders by ID" in new DbContext{
+    val catOrder: Order = Order(None, Some(8), Some(100), Some("2015-07-01T17:36:58.190Z"), Some(Placed), Some(false))
+    val idFuture: Future[Long] = db.addOrder(catOrder)
+    val success: Future[Boolean] = for{
+      id <- idFuture
+    } yield Await.result(db.deleteOrder(id))
+    Await.result(success)
+  }
+
+  it should "fail appropriately when trying to delete nonexistent orders by ID" in new DbContext {
+    assert(!Await.result(db.deleteOrder(100)))
+  }
+
+  //GET: Find purchase order by ID
+  it should "be able to find an order by its ID" in new DbContext{
+    check{ (order: Order) =>
+      val inputOrder: Order = order.copy(id = None)
+      val idFuture: Future[Long] = db.addOrder(inputOrder)
+      val retOrder: Future[Order] = db.findOrder(Await.result(idFuture))
+      Await.result(retOrder).equals(inputOrder.copy(id = Some(Await.result(idFuture))))
+     }
+  }
+
+  //============================STORE TESTS END HERE================================================
+
+  //+++++++++++++++++++++++++++++USER TESTS BEGIN HERE+++++++++++++++++++++++++++++++++++++++++
+
+  //POST: Create user
+  it should "be able to add new users" in new DbContext{
+    check{ (u: User) =>
+      val inputUser: User = u.copy(id = None)
+      val name: String = Await.result(db.addUser(inputUser))
+      Await.result(db.getUser(name)).copy(id = None) === inputUser
+     }
+  }
+
+  it should "fail when trying to create a new user that's passed with an ID" in new DbContext {
+    val inputUser: User = User(Some(1), "theCat", Some("Cat"), None, None,
+      "butyouarewrong", None)
+    Await.result(db.addUser(inputUser).liftToTry).isThrow
+  }
+
+  //DELETE: Delete user
+  it should "be able to delete users" in new DbContext {
+    Await.ready(db.deleteUser("coraline"))
+  }
+
+  //GET: Get user by username, assume all usernames are unique
+  it should "facillitate searching for users via username" in new DbContext{
+    val eSwan: User = User(None, "pirateKing", Some("Elizabeth"), Some("Swan"), Some("eswan@potc.com"),
+      "hoistTheColours", None)
+    db.addUser(eSwan)
+    assert(Await.result(db.getUser("pirateKing")).copy(id = None).equals(eSwan))
+  }
+
+  //PUT: Update user
+  it should "allow the updating of existing users" in new DbContext{
+    val updateMe: User = User(None, "chrysanthemum", Some("Jay"), Some("Chou"), Some("jchou@jay.com"),
+    "terrace", None)
+    db.addUser(updateMe)
+    check{ (u:User) =>
+      val inputUser: User = u.copy(id = None, username = updateMe.username)
+      Await.result(db.updateUser(inputUser)).copy(id = None).equals(inputUser)
+     }
+  }
+
+  //============================USER TESTS END HERE================================================
```

Comment:
We said we'd get rid of them in all the non-test files but leave them around in the test files. Would it be better to get rid of them though?